client/__tests__/e2e/e2e.test.ts

```diff
@@ -85,448 +78,477 @@ const setToRemoveFrom = "empty_this_geneset";
     );
 
-    await drag("layout-graph", lassoSelection.start, lassoSelection.end, true);
+    await drag(
+      "layout-graph",
+      lassoSelection.start,
+      lassoSelection.end,
+      page,
+      true
+    );
 
-    const cellCount = await getCellSetCount(1);
+    const cellCount = await getCellSetCount(1, page);
     expect(cellCount).toBe(data.subset.lasso.count);
   });
 });
 
 describe("clipping", () => {
-  test("clip continuous", async () => {
-    await goToPage(pageUrl);
-
-    await clip(data.clip.min, data.clip.max);
+  test("clip continuous", async ({ page }) => {
+    await goToPage(page);
+    await clip(data.clip.min, data.clip.max, page);
     const histBrushableAreaId = `histogram-${data.clip.metadata}-plot-brushable-area`;
     const coords = await calcDragCoordinates(
       histBrushableAreaId,
-      data.clip["coordinates-as-percent"]
+      data.clip["coordinates-as-percent"],
+      page
     );
-    await drag(histBrushableAreaId, coords.start, coords.end);
-    const cellCount = await getCellSetCount(1);
+    await drag(histBrushableAreaId, coords.start, coords.end, page);
+    const cellCount = await getCellSetCount(1, page);
     expect(cellCount).toBe(data.clip.count);
 
     // ensure categorical data appears properly
-    for (const label of Object.keys(data.categorical)) {
-      const element = await getOneElementInnerHTML(
-        getTestId(`category-${label}`)
-      );
+    for (const label of Object.keys(
+      data.categorical
+    ) as (keyof typeof data.categorical)[]) {
+      const element = await page.getByTestId(`category-${label}`).innerHTML();
 
       expect(element).toMatchSnapshot();
 
-      await clickOn(`${label}:category-expand`);
+      await page.getByTestId(`${label}:category-expand`).click();
 
-      const categories = await getAllCategoriesAndCounts(label);
+      const categories = await getAllCategoriesAndCounts(label, page);
 
       expect(Object.keys(categories)).toMatchObject(
-        // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
         Object.keys(data.categorical[label])
       );
 
       expect(Object.values(categories)).toMatchObject(
-        // @ts-expect-error ts-migrate(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
         Object.values(data.categorical[label])
       );
-    }    
+    }
   });
 });
 
 // interact with UI elements just that they do not break
 describe("ui elements don't error", () => {
-  test("color by", async () => {
-    await goToPage(pageUrl);
-
+  test("color by", async ({ page }) => {
+    await goToPage(page);
     const allLabels = [
       ...Object.keys(data.categorical),
       ...Object.keys(data.continuous),
     ];
 
     for (const label of allLabels) {
-      await clickOn(`colorby-${label}`);
+      await page.getByTestId(`colorby-${label}`).click();
```

Comment:
Maybe worth a ticket to capture this, but this and the original test don't seem to assert after clicking the element it does color the umap 😆

Reply:
#716